pkg/schedule/scatter/region_scatterer.go

```diff
@@ -340,9 +326,15 @@ func (r *RegionScatterer) scatterRegion(region *core.RegionInfo, group string, s
 				// It is both sourcePeer and targetPeer itself, no need to select.
 				continue
 			}
+			sourceStore := r.cluster.GetStore(peer.GetStoreId())
+			if sourceStore == nil {
+				log.Error("failed to get the store", zap.Uint64("store-id", peer.GetStoreId()), errs.ZapError(errs.ErrGetSourceStore))
+				continue
+			}
+			filters[filterLen-1] = filter.NewPlacementSafeguard(r.name, r.cluster.GetSharedConfig(), r.cluster.GetBasicCluster(), r.cluster.GetRuleManager(), region, sourceStore, oldFit)
 			for {
-				candidates := r.selectCandidates(region, oldFit, peer.GetStoreId(), selectedStores, context)
-				newPeer := r.selectStore(group, peer, peer.GetStoreId(), candidates, context)
+				filters[filterLen-2] = filter.NewExcludedFilter(r.name, nil, selectedStores)
```

Comment:
I think this filter is same for all peers

Reply:
selectedStores is different .

Reply:
selectedStores is a pointer

Reply:
yes, fixed